src/PrestaShopBundle/ApiPlatform/Provider/QueryProvider.php

```diff
@@ -31,20 +31,28 @@
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
 use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShopBundle\ApiPlatform\Exception\NoExtraPropertiesFoundException;
 
 class QueryProvider implements ProviderInterface
 {
-    private $queryBus;
-
-    public function __construct(CommandBusInterface $queryBus)
+    public function __construct(private CommandBusInterface $queryBus)
     {
-        $this->queryBus = $queryBus;
     }
 
     public function provide(Operation $operation, array $uriVariables = [], array $context = [])
     {
-        $query = $operation->getExtraProperties()['query'];
+        $extraProperties = $operation->getExtraProperties();
+
+        $query = array_key_exists('query', $extraProperties) ? $extraProperties['query'] : null;
+        $command = array_key_exists('command', $extraProperties) ? $extraProperties['command'] : null;
```

Comment:
Are we going to process commands and queries in the `QueryProvider` ?

Reply:
Well actually you're right. We talked about it with @jolelievre and no there is a different provider for the commands. In the future we will maybe merge the both of them but not for now. Will remove it